pkg/controller/operators/catalog/subscription/syncer.go

```diff
@@ -67,9 +69,21 @@ func (s *subscriptionSyncer) Sync(ctx context.Context, event kubestate.ResourceE
 	case kubestate.ResourceAdded:
 		initial = initial.Add()
 	case kubestate.ResourceUpdated:
+		oldSubscription, err := s.client.OperatorsV1alpha1().Subscriptions(res.GetNamespace()).Get(context.TODO(), res.GetName(), metav1.GetOptions{})
+		if err != nil {
+			logger.Errorf("Could not get old subscription %s in namespace %s. Skipping attempt to delete metrics", oldSubscription.GetName(), oldSubscription.GetNamespace())
+		} else {
+			if res.GetName() != oldSubscription.GetName() ||
```

Comment:
`res.GetName() != oldSubscription.GetName()` should never happen, right?

Also, based on https://github.com/operator-framework/operator-lifecycle-manager/pull/1519/files#r431776633, I think we want to make sure `currentCSV != installedCSV` before deleting.

@ecordell does this line up with your comment?